### ISSUE TITLE
[BUG] Loading quantized GGUF models fails in 0.11.0

### ISSUE THREAD

**Issue by armbues** [ml-explore/mlx#1012]
**Describe the bug**
Loading weights from GGUF files fails for quantized models after updating to version 0.11.0:
`ValueError: [eval] Attempting to eval an array without a primitive.`

**To Reproduce**

Include code snippet
```python
import mlx.core as mx

weights, metadata = mx.load(gguf_path , return_metadata=True)
mx.eval(weights)
```

**Expected behavior**

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.4.1
 - Version 0.11.0

**Additional context**
I've tested this with multiple GGUF model files that worked before and now result in the error message.

Here's a model that definitely worked before and isn't loading anymore: [Llama-2-7B-Chat.Q8_0.gguf](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF/blob/main/llama-2-7b-chat.Q8_0.gguf)

**Comment by awni** [comment]
Sorry that broke from 0.10 to 0.11. Fix is in #1014, you can revert to the old version until the next release.

**Comment by armbues** [comment]
Awesome, thank you!
